.github/workflows/go.yml

```diff
@@ -0,0 +1,177 @@
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Set up Go ${{ matrix.go }}
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ matrix.go }}
+
+      - name: Install shared software dependencies
+        uses: ./.github/workflows/install-shared-dependencies
+        with:
+          os: ${{ matrix.os }}
+          target: ${{ matrix.os == 'ubuntu-latest' && 'x86_64-unknown-linux-gnu' || 'x86_64-apple-darwin' }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install redis
+        # TODO: make this step macos compatible: https://github.com/aws/glide-for-redis/issues/781
+        if: ${{ matrix.os == 'ubuntu-latest' }}
+        uses: ./.github/workflows/install-redis
+        with:
+          redis-version: ${{ matrix.redis }}
+
+      - name: Install client dependencies
+        working-directory: ./go
+        run: make install-tools
+
+      - name: Build client
+        working-directory: ./go
+        run: |
+          go mod edit -go=${{ matrix.go }}
+          make build
+
+      - name: Run go vet linter
+        working-directory: ./go
+        run: go vet .
+
+      - name: Run staticcheck linter
+        working-directory: ./go
+        run: staticcheck .
+
+      - name: Run unit tests
+        working-directory: ./go
+        run: make unit-test-report
+
+      - name: Upload test reports
+        if: always()
+        continue-on-error: true
+        uses: actions/upload-artifact@v4
+        with:
+          name: test-reports-go-${{ matrix.go }}-redis-${{ matrix.redis }}-${{ matrix.os }}
+          path: |
+            go/reports/unit-test-report.html
+
+  build-amazonlinux-latest:
+    if: github.repository_owner == 'aws'
+    strategy:
+      # Run all jobs
+      fail-fast: false
+      matrix:
+        go:
+          - 1.18.10
+          - 1.21.6
+    runs-on: ubuntu-latest
+    container: amazonlinux:latest
+    timeout-minutes: 15
+    steps:
+      - name: Install git
+        run: |
+          yum -y remove git
+          yum -y remove git-*
+          yum -y install https://packages.endpointdev.com/rhel/7/os/x86_64/endpoint-repo.x86_64.rpm
+          yum update
+          yum install -y git
+          git --version
+
+      - uses: actions/checkout@v4
+
+      - name: Checkout submodules
+        run: |
+          git config --global --add safe.directory "$GITHUB_WORKSPACE"
+          git submodule update --init --recursive
+
+      - name: Install shared software dependencies
+        uses: ./.github/workflows/install-shared-dependencies
+        with:
+          os: "amazon-linux"
+          target: "x86_64-unknown-linux-gnu"
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Create a symbolic Link for redis6 binaries
+        run: |
+          ln -s /usr/bin/redis6-server /usr/bin/redis-server
+          ln -s /usr/bin/redis6-cli /usr/bin/redis-cli
+
+      - name: Install Go
```

Comment:
Can't you use `actions/setup-go` there?

Reply:
Maybe? I saw that Java uses setup-java for ubuntu/macos but not for amazon linux so I assumed setup-java did not work on amazon linux and that if it doesn't work for java that it wouldn't for go. But my assumption may be wrong